File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        max_profit = 0
        buy = prices[0]
        for i in range(1, len(prices)):
            if(buy>prices[i]):
                buy = prices[i]
            elif(prices[i]-buy > max_profit):
                max_profit = prices[i]-buy

        return max_profit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let maxProfit = 0
    let minValue = prices[0]
    for (let i = 1; i < prices.length; i++) {
        minValue = prices[i] < minValue ? prices[i]: minValue
        maxProfit = (prices[i] - minValue) > maxProfit ? prices[i]-minValue : maxProfit 
    }
    return maxProfit
};